### ISSUE TITLE
GPU feature cannot be compiled without MPI

### ISSUE THREAD

**Issue by wdeconinck** [ecmwf-ifs/ectrans#138]
The current GPU code-path cannot be compiled without MPI.
This is an issue if we want to use ectrans with GPU in a MPI-serial context.

**Comment by samhatfield** [comment]
The solution is to move to MPL for calls like [this](https://github.com/ecmwf-ifs/ectrans/blob/main/src/trans/gpu/internal/trgtol_mod.F90#L583).

I think @marsdeno has done some work on this?

**Comment by reuterbal** [comment]
Indeed. This was blocked by support for GPU-aware MPI communication in MPL, which in turn I believe requires using mpi_f08 in MPL.

**Comment by wdeconinck** [comment]
Arguably in a context of ENABLE_MPI=OFF (serial) there's nothing to send or receive and mpi_f08 would not be needed either.
Preprocessor guards could be added where needed temporarily, even if it's ugly, until MPL can handle the GPU-aware comms.

**Comment by wdeconinck** [comment]
Resolved with #140
